### ISSUE TITLE
Bug: [no-magic-numbers] false positive on <N> when ignoreNumericLiteralTypes is true

### ISSUE THREAD

**Issue by a-non-a-mouse** [typescript-eslint/typescript-eslint#11821]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.2&showAST=es&fileType=.tsx&code=MYewdgzgLgBARgQwF4wLwygCwJZgOYA8ALAEwB8AFAJQDcMA9PTAKYBOrIrQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oFsBDAc0plaTWNwBGiaKgwBtSFOgdokADRZsOSv07REAOTFShAGUr4pveABUSKdGHxxE4bAF8Auq7cg3QA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
type foo = 42; // no error
const baz = thing<42>(); // error
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  "rules": {
    "@typescript-eslint/no-magic-numbers": ["error",{
      "ignoreNumericLiteralTypes": true
    }]
  }
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

`<42>` is a numeric literal type.  It shouldn't trigger an error when ignoreNumericLiteralTypes is true

### Actual Result

incorrectly triggers an error

**Comment by kirkwaiblinger** [comment]
Hi @a-non-a-mouse !

It looks like this has been the case since the initial implementation, in https://github.com/typescript-eslint/typescript-eslint/pull/373. 

I think the idea, though not well documented, is that the option treats numeric literals like ordinary numbers, and insists that they must be named in order to be used. Consider the following:

([playground](https://typescript-eslint.io/play/#ts=5.9.2&showAST=es&fileType=.tsx&code=CYUwxgNghgTiAEAzArgOzAFwJYHtXwwAstUBzAHgBV4BeeKVATwD4AKADwH4AueVZALYAjEDACUvBowDcAKFlESFACwAmNmOnwA9NvgCopLGD6CRMeRkYAHBADEcMK6trw1W3fBDtrEY1gwIRj4oARBgU2FRBWIycgcnRnVWTXlPBigIHFIcZABnAhxLWNJWNU0dPQMjE34oi1kwPDyMJEdnV3dKrx8-MACgkLCIuvMYpVZEdqTNIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oFsBDAc0plaTWNwBGiaKgwBtSAHde0JpAA0WbDkr9O0RADkxUoQBlK%2BKb3gAVEinRh8cROGwBfALpv3Id0A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false))

```ts
declare function thing<T = any>(x?: number): any;

thing<42>(); // magic number

type Forty2 = 42; // explicitly named number
thing<Forty2>();

// analogous to

thing(42); // magic number

const forty2 = 42; // explicitly named number
thing(forty2);
```

**Comment by kirkwaiblinger** [comment]
Closing in favor of conversation at https://github.com/eslint/eslint/issues/20372.
